doc/source/tutorial.rst

```diff
@@ -0,0 +1,284 @@
+Tutorial
+========
+
+To use Ray, you need to understand the following:
+
+- How Ray uses object IDs to represent immutable remote objects.
+- How Ray executes tasks asynchronously to achieve parallelism.
+
+Overview
+--------
+
+Ray is a Python-based distributed execution engine. It can be used on a single
+machine to achieve efficient multiprocessing, and it can be used on a cluster
+for large computations.
+
+When using Ray, several processes are involved.
+
+- Multiple **worker** processes execute tasks and store results in object
+  stores. Each worker is a separate process.
+- One **object store** per node stores immutable objects in shared memory and
+  allows workers to efficiently share objects on the same node with minimal
+  copying and deserialization.
+- One **local scheduler** per node assigns tasks to workers on the same node.
+- A **global scheduler** receives tasks from local schedulers and assigns them
+  to other local schedulers.
+- A **driver** is the Python process that the user controls. For example, if the
+  user is running a script or using a Python shell, then the driver is the Python
+  process that runs the script or the shell. A driver is similar to a worker in
+  that it can submit tasks to its local scheduler and get objects from the object
+  store, but it is different in that the local scheduler will not assign tasks to
+  the driver to be executed.
+- A **Redis server** maintains much of the system's state. For example, it keeps
+  track of which objects live on which machines and of the task specifications
+  (but not data). It can also be queried directly for debugging purposes.
+
+Starting Ray
+------------
+
+To start Ray, start Python and run the following commands.
+
+.. code-block:: python
+
+  import ray
+  ray.init()
+
+This starts Ray along with a number of other processes including local and
+global schedulers, an object store, an object manager, a Redis server, and a
+bunch of worker processes. These will be killed when you exit the Python
+interpreter.
+
+Immutable remote objects
+------------------------
+
+In Ray, we can create and manipulate objects. We refer to these objects as
+**remote objects**, and we use **object IDs** to refer to them. Remote objects
+are stored in **object stores**, and there is one object store per node in the
+cluster. In the cluster setting, we may not actually know which machine each
+object lives on.
+
+An **object ID** is essentially a unique ID that can be used to refer to a
```

Comment:
This paragraph feels out of place, since we haven't mentioned tasks yet. Perhaps another way of saying this is that you can have a reference to an object ID whose value hasn't been written yet.